ledger/src/blockstore.rs

```diff
@@ -1500,19 +1503,43 @@ impl Blockstore {
 
     pub fn read_transaction_status(
         &self,
-        index: (Slot, Signature),
+        index: (Signature, Slot),
     ) -> Result<Option<RpcTransactionStatusMeta>> {
         self.transaction_status_cf.get(index)
     }
 
     pub fn write_transaction_status(
         &self,
-        index: (Slot, Signature),
+        index: (Signature, Slot),
         status: &RpcTransactionStatusMeta,
     ) -> Result<()> {
         self.transaction_status_cf.put(index, status)
     }
 
+    fn prune_transaction_status_cf_by_slot(
+        &self,
+        batch: &mut WriteBatch,
+        from_slot: Slot,
+        to_slot: Slot,
+    ) -> Result<bool> {
+        let mut results: Vec<bool> = vec![];
```

Comment:
I think you can change this to `(from_slot..to_slot).iter().map()...collect()`

Reply:
If I remove that `?` try, then yes ;)